fs/src/jvmMain/kotlin/com/suparnatural/core/fs/FileSystem.kt

```diff
@@ -0,0 +1,228 @@
+package com.suparnatural.core.fs
+
+import java.io.*
+import java.nio.file.Paths
+import java.util.*
+
+actual object FileSystem {
+
+    actual val contentsDirectory: Path
+        get() {
+            val path = Paths.get("./contents/").toAbsolutePath().toString()
+            File(path).mkdirs()
+            return Path.simplified(path)
+        }
+
+
+    actual val cachesDirectory: Path
+        get() {
+            val path = Paths.get("./caches/").toAbsolutePath().toString()
+            File(path).mkdirs()
+            return Path.simplified(path)
+        }
+
+
+    actual val temporaryDirectory: Path
+        get() = cachesDirectory
+
+
+    private fun buildStats(file: File): StatResult {
+        val fileType = {
+            when {
+                file.isDirectory -> FileType.Directory
+                file.isFile -> FileType.Regular
+                else -> FileType.Unknown
+            }
+        }()
+        return StatResult(
+                name = file.name,
+                absolutePath = PathComponent(file.absolutePath),
+                canonicalPath = PathComponent(file.canonicalPath),
+                createdAt = 0.0,
+                modifiedAt = file.lastModified().toDouble(),
+                size = file.length().toDouble(),
+                type = fileType
+        )
+    }
+
+    actual fun readDir(path: String): List<StatResult>? {
+        val parent = File(path).canonicalFile ?: return null
+        val fileList = parent.listFiles() ?: return null
+        return fileList.map {
+            buildStats(it)
+        }
+    }
+
+
+    actual fun readDir(pathComponent: PathComponent): List<StatResult>? {
+        val path = pathComponent.component ?: return null
+        return readDir(path)
+    }
+
+
+    actual fun stat(path: String): StatResult? {
+        val file = File(path)
+        return buildStats(file)
+    }
+
+
+    actual fun stat(pathComponent: PathComponent): StatResult? {
+        val path = pathComponent.component ?: return null
+        return stat(path)
+    }
+
+
+    actual fun readFile(path: String, encoding: ContentEncoding): String? {
+        val file = File(path).canonicalFile
+
+        val charset = {
+            when (encoding) {
+                ContentEncoding.Ascii -> Charsets.US_ASCII
+                else -> Charsets.UTF_8
+            }
+        }()
+        val reader = BufferedReader(InputStreamReader(FileInputStream(file), charset))
+        val content = reader.readLines().joinToString("\n")
```

Comment:
There should be some IOException handling. If reader throw exception, stream will not be closed and will leak.

E.g. reader.use {
    content = reader.readLines().joinToString("\n")
} //will close reader automatically and ignore exception (if that was intended)